### PR TITLE
incoming irc message tags support

### PR DESCRIPTION
**Pull request by ailin-nemui**
this is just the very bare bones to accept and ignore incoming
tags. further patches can build upon it

original implementation by @dequis #576

### CODE REVIEW

**Comment by dequis** [comment]
Reminds me of #576 :D

**Comment by ailin-nemui** [comment]
oops

**Comment by ailin-nemui** [comment]
in #576, I wrote

> I would like to see actual use case implementation before proceeding here...

now, I'm posting the same patch again but I'm already working on use cases. Separate patch for easier review

**Comment by dequis** [comment]
Yeah no worries, I didn't really have the time to do the whole thing (and still don't)

File: src/irc/core/irc.c [hunk 1]
```diff
@@ -382,14 +406,14 @@ static char *irc_parse_prefix(char *line, char **nick, char **address)
...
+	char *nick, *address, *tags;
 
 	g_return_if_fail(server != NULL);
 	g_return_if_fail(line != NULL);
 
-	line = irc_parse_prefix(line, &nick, &address);
-	if (*line != '\0')
-		signal_emit_id(signal_server_event, 4, server, line, nick, address);
+	line = irc_parse_prefix(line, &nick, &address, &tags);
+	if (*line != '\0' || tags != NULL)
```
Comment (dequis) [inline src/irc/core/irc.c]: So wait, this is handling tags and no line like...

    @tags \r\n

I don't think that's a valid irc message (the command is always required) or something that makes sense to have special cases for.

File: src/irc/core/irc.c [hunk 2]
```diff
@@ -158,6 +159,11 @@ static char *split_nicks(const char *cmd, char **pre, char **nicks, char **post,
 
 	*pre = g_strdup(cmd);
 	*post = *nicks = NULL;
+
+	if (**pre == '@') { /* message-tags */
+		arg++;
```
Comment (dequis) [inline src/irc/core/irc.c]: This one took me a while to understand, could use a longer comment

File: src/irc/core/irc.c [hunk 3]
```diff
@@ -441,12 +465,14 @@ static void irc_init_server(IRC_SERVER_REC *server)
 void irc_irc_init(void)
 {
 	signal_add("server event", (SIGNAL_FUNC) irc_server_event);
+	signal_add("server event tags", (SIGNAL_FUNC) irc_server_event_tags);
```
[the remark this answers is not in the document]
Comment (ailin-nemui) [reply src/irc/core/irc.c]: Good catch

File: src/irc/core/irc.c (hunk 1 above)
Comment (ailin-nemui) [reply src/irc/core/irc.c]: wouldn't it be better to preserve and make accessible such an invalid line nevertheless?

File: src/irc/core/irc.c (hunk 2 above)
Comment (ailin-nemui) [reply src/irc/core/irc.c]: I added a comment

Comment (dequis) [reply src/irc/core/irc.c]: Sounds good

File: src/irc/core/irc.c (hunk 1 above)
Comment (dequis) [reply src/irc/core/irc.c]: Yeah, I don't know. I guess it doesn't really make other things more complicated and it's harmless to do so.

**Pull request merged**